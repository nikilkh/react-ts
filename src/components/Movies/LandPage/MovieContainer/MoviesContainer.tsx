import { useContext } from "react";
import { context, movieData } from "../../../context";
import { MoviesContainerStyle, MovieStyle } from "../LandPage.styled";
import MovieCard from "./MovieCard";
const IMG_URL = "https://image.tmdb.org/t/p/w500";

export default function MoviesContainer() {
  const states = useContext(context);

  return (
    <MoviesContainerStyle>
      {states?.showMovies ? (
        <>
          {states?.movies?.map((movie: movieData) => {
            return (
              <>
                <MovieStyle bg={IMG_URL + movie.poster_path}>
                  <MovieCard movie={movie}></MovieCard>
                </MovieStyle>
              </>
            );
          })}
        </>
      ) : (
        <h3>loading...</h3>
      )}
    </MoviesContainerStyle>
  );
}
