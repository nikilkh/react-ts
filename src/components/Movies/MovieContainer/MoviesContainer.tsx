import { useContext } from "react";
import { context, movieData } from "../../Context/context";
import { MoviesContainerStyle, MovieStyle } from "../LandPage/LandPage.styled";
import { MovieCard } from "../MovieCard/index";
const IMG_URL = "https://image.tmdb.org/t/p/w500";

export const MoviesContainer = () => {
  const states = useContext(context);

  return (
    <MoviesContainerStyle>
      {states?.showMovies ? (
        <>
          {states?.movies?.map((movie: movieData) => {
            return (
              <>
                <MovieStyle bg={IMG_URL + movie.poster_path}>
                  <MovieCard movie={movie}></MovieCard>
                </MovieStyle>
              </>
            );
          })}
        </>
      ) : (
        <h3>loading...</h3>
      )}
    </MoviesContainerStyle>
  );
}
