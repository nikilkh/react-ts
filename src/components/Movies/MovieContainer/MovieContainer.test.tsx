import { render } from "@testing-library/react";
import { MoviesContainer } from "./MoviesContainer";

describe("renders learn react link", () => {
  it("should render hello world", () => {
    const {container} = render(<MoviesContainer/>);
    expect(container).toMatchSnapshot();
  });
});