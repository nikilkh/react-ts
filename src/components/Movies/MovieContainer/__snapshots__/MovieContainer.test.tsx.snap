// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link should render hello world 1`] = `
<div>
  <div
    class="sc-jSMfEi iAkGmp"
  >
    <h3>
      loading...
    </h3>
  </div>
</div>
`;
